### ISSUE TITLE
[FEATURE REQUEST] Add opportunity to start showing the mask on focus

### ISSUE THREAD

**Issue by akhmadullin** [s-yadav/react-number-format#710]
## Intro
Hi!

First of all thanks for a great lib!

## Problem
I want to show placeholder, when input is empty. Also I want to show mask formatting, when input is focused, even if it is empty.

Now I can use prop `allowEmptyFormatting` to show mask, when input is empty, but mask is shown constantly, even if input is not focused. Also active `allowEmptyFormatting` prop hides placeholder, because of permanent mask showing.

[Demo with problem explanation](https://codesandbox.io/s/allow-empty-formatting-and-placeholder-demo-4g8yhc?file=/src/App.js)

## Feature request
I suggest to add opportunity to
- show placeholder, when input is empty and not focused
   
<img width="163" alt="Снимок экрана 2022-12-06 в 14 40 00" src="https://user-images.githubusercontent.com/37641303/205819183-c9873425-e281-462a-97c3-3b70e76ab529.png">

- show mask, when input is empty and focused
   
<img width="163" alt="Снимок экрана 2022-12-06 в 14 40 18" src="https://user-images.githubusercontent.com/37641303/205819207-861dda21-bbec-4314-9bdd-a9687b3b4ddf.png">

- show mask (formatted value), when input is filled (no matter focused it or not)
   
<img width="163" alt="Снимок экрана 2022-12-06 в 14 53 45" src="https://user-images.githubusercontent.com/37641303/205819250-6a64f321-0740-470f-809f-e146d0a94512.png">
<img width="163" alt="Снимок экрана 2022-12-06 в 14 53 47" src="https://user-images.githubusercontent.com/37641303/205819266-7a1eeca3-3635-40cc-bd16-f33cfa5dc70d.png">

Is it possible to implement?

**Comment by s-yadav** [comment]
This can be implement outside of the lib.
```js
function MyNumericFormat(props) {
  const [onFocus, toggleFocus] = useState();
  return (
     <PatternFormat
          {...props}
          onFocus={() => toggleFocus(true)}
          onBlur={() => toggleFocus(false)}
          allowEmptyFormatting={onFocus}
        />
  )
}
```
Update codesandbox for the above. https://codesandbox.io/s/allow-empty-formatting-and-placeholder-demo-forked-l5f3kj

**Comment by akhmadullin** [comment]
Yes, I also tried this approach with using of outside `useState`. 

But changing of `useState` value triggers additional rerender - it seems not to be very efficient/performant.

Approach with using of outside `useState` looks like a hacky way to implement desired behavior.

If `PatternFormat` was able to add mask on focus under the hood, it would be great and more performant.

Is it possible to implement such feature inside lib?
How difficult will it be to do this?

**Comment by s-yadav** [comment]
Ohhh, if you create a wrapper component of PatternFormat and do the state management there, the rerender will be localised to that wrapper. The rerender cost would be negligible to worry about.

As the behaviour can be implemented outside, will avoid it to introduce in the lib, as it will increase the footprint and maintenance cost in future.
